RFC-001-BAM.md

```diff
@@ -166,127 +169,92 @@ Ids MAY be skipped but MUST be ascending.
+Application protocols can be described by defining `REQUEST` types and with them, the semantics of certain headers of a `REQUEST`.
 
 In addition, headers also encode compatibility information.
 Each header is available in two variants:
 
 - MUST understand
 - MAY ignore
 
-If a node receives a header in the `MUST understand` variant in a `REQUEST` and it does not understand it, it MUST reject the request with a `SE01` response.
-See section [Sender Error Responses](#sender-errors-se00-se99) for further details.
+If a node receives a header in the `MUST understand` variant in a `REQUEST` and it does not understand it, it MUST reject the request with an `ERROR` frame of type `unknown-mandatory-header`.
 Headers encoded as `MAY ignore` are ok to be not understood.
 Nodes may simply ignore them as if they were not there.
 
-If a node receives a `RESPONSE` frame with a header that is marked as `MUST understand` but it does not understand it, it MUST stop processing this response.
-The reason behind this is that the responding party demanded understanding of this header in order to continue.
-This is most likely due to a backwards-incompatible change in a protocol where the receiving party would act in an incompatible way if it did not take the mandatory header into account.
-
-##### Status code families
-
-Similar to HTTP, this protocol divides status codes into different families.
-By separating semantically different responses from each other, programs are able understand these semantics and act upon them.
-The simplest example is probably successful versus non-successful responses.
-Even within non-successful responses, one can distinguish between errors caused by the sender and errors caused by the Receiver (400 vs 500 in HTTP).
-
-The purpose of status codes is to allow nodes to act in pre-defined ways for certain error/success scenarios.
-For example, a node could be configured auto-retry a request for certain errors if it knows how to resolve them.
-
-In order to avoid confusion with HTTP, the status code families are defined as a pair of characters.
-Individual codes within the family have a particular meaning.
-
-- OK00 (read as OK-00)
-- SE00 (read as **S**ender**E**rror-00)
-- RE00 (read as **R**eceiver**E**rror-00)
-
-This protocol reserves the status codes 00 to 19 for all of the families.
-Application protocols that build on top of this transport protocol MUST NOT redefine any status codes within this range.
-However, they are highly encouraged to define status codes started from `20` for their own purposes.
+To avoid more complexity through additional messages, the spec doesn't define a concept of acknowledging successful processing of a `RESPONSE` to the sender.
+Thus, there is no way of signaling to the sender of a `RESPONSE` whether or not it was properly understood.
+Therefore, careful thought should be put into the design of and use of the `MUST understand` variant of a header to make this failure case as rare as possible.
```

Comment:
I am not sure I am following. if there is a `MUST understand` that is not understand then the other node will send an `ERROR` frame, right?

How is it related to lack of acknowledgement type frames?

Reply:
`ERROR` frames are an alternative to `RESPONSE` frames. There are no "responses" to invalid `RESPONSE` frames.